orttraining/orttraining/python/training/api/checkpoint_state.py

```diff
@@ -52,33 +101,52 @@ def save_checkpoint(
         """
         C.save_checkpoint(state._state, os.fspath(checkpoint_uri), include_optimizer_state)
 
-    def __getitem__(self, name: str) -> int | float | str:
-        """Gets the property associated with the given name
+    def __getitem__(self, name: str) -> int | float | str | Parameter:
+        """Gets the parameter or property associated with the given name
+
+        Searches for the name in the parameters and properties of the checkpoint state.
```

Comment:
Why didn't we design the Parameter class to handle the process of getting and setting parameters directly, and have instances of the Parameter class as attributes within the Module or CheckpointState?

Reply:
I think this is a fair question. Since the properties was already exposed as a indexable attribute of the CheckpointState, I extended the idea to parameters as well.

Basically, properties and parameters are bot indexable with the checkpoint state with my current approach. But maybe that does not add a lot of value. I can probably remove the CheckpointState from being indexable, and add parameters and properties as attributes of the checkpoint state (and they can be indexable).
Will change this.